### PR TITLE
Rewritten unit tests for highlighting issues with some French banks

### PR DESCRIPTION
**Pull request by ambroisemaupate**
I’ve rewritten PHPUnit tests to be able to test more effectively with as many OFX samples as wanted. And I included *PHPUnit* in *Composer* dependancies. I‘ve provided OFX samples for:

- `bpbfc` (Banque Populaire)
- `cmfr` (Crédit Mutuel)
- `oneline` - sample with no line-breaks

Parsing is still faulty with *bpbfc* as this file mixes unclosed XML tags and well-closed tags. (lines 33-36).

- `cmfr` and `oneline` pass the tests.
- **Make sure to launch *phpunit* with php 5.6 maximum**

This PR highlights some issue with PR #22 and `oneline` sample.

Thanks for your review!
👍

### CODE REVIEW

File: tests/OfxParser/OfxTest.php [hunk 1]
```diff
@@ -19,9 +19,12 @@ public function setUp()
         $this->ofxdata = simplexml_load_string( file_get_contents($ofxFile) );
     }
 
+    /**
```
Comment (asgrim) [inline tests/OfxParser/OfxTest.php]: Please could you use `$this->expectException(..)` instead of annotations?

File: composer.json [hunk 2]
```diff
@@ -22,6 +22,9 @@
     "require": {
         "php": ">=5.4"
     },
+    "require-dev": {
+        "phpunit/phpunit": "~4.1.3"
```
Comment (asgrim) [inline composer.json]: Why not require the most recent phpunit?

File: tests/OfxParser/ParserTest.php [hunk 3]
```diff
@@ -50,35 +43,77 @@ public function testXmlLoadStringLoadsValidXml()
...
-		$this->assertEquals('<SOMETHING>', $method->invoke($parser, '<SOMETHING>'));
-		$this->assertEquals('<SOMETHING>foo</SOMETHING>', $method->invoke($parser, '<SOMETHING>foo'));
-		$this->assertEquals('<SOMETHING>foo</SOMETHING>', $method->invoke($parser, '<SOMETHING>foo</SOMETHING>'));
+		$this->assertEquals($expected, $method->invoke($parser, $input));
 	}
 
-	public function testConvertSgmlToXml()
-	{
-		$sgml = <<<HERE
+	public function testConvertSgmlToXmlProvider()
```
Comment (asgrim) [inline tests/OfxParser/ParserTest.php]: Please use spaces for indentation consistently, there's a mixture of tabs/spaces going on here

File: composer.json (hunk 2 above)
Comment (ambroisemaupate) [inline composer.json]: Yes, I can increase it to 4.8 maximum to be compatible with PHP 5.4 and 5.5

Comment (asgrim) [inline composer.json]: PHP 5.4 and 5.5 are not supported versions of PHP, so happy to go to minimum 5.6.

**Comment by asgrim** [comment]
@ambroisemaupate I'm getting a test failure `OfxParser\ParserTest::testLoadFromString with data set #3 ('/home/james/workspace/ofxpars...fc.ofx')`

```
$ vendor/bin/phpunit tests
PHPUnit 5.5.2 by Sebastian Bergmann and contributors.

...........................E                                      28 / 28 (100%)

Time: 46 ms, Memory: 4.00MB

There was 1 error:

1) OfxParser\ParserTest::testLoadFromString with data set #3 ('/home/james/workspace/ofxpars...fc.ofx')
Exception: Failed to parse OFX: array (
  0 => 
  LibXMLError::__set_state(array(
     'level' => 3,
     'code' => 76,
     'column' => 14,
     'message' => 'Opening and ending tag mismatch: BANKACCTFROM line 45 and BANKID
',
     'file' => '',
     'line' => 48,
  )),
  1 => 
  LibXMLError::__set_state(array(
     'level' => 3,
     'code' => 76,
     'column' => 12,
     'message' => 'Opening and ending tag mismatch: STMTRS line 41 and BRANCHID
',
     'file' => '',
     'line' => 51,
  )),
  2 => 
  LibXMLError::__set_state(array(
     'level' => 3,
     'code' => 76,
     'column' => 10,
     'message' => 'Opening and ending tag mismatch: STMTTRNRS line 29 and ACCTID
',
     'file' => '',
     'line' => 54,
  )),
  3 => 
  LibXMLError::__set_state(array(
     'level' => 3,
     'code' => 76,
     'column' => 12,
     'message' => 'Opening and ending tag mismatch: BANKMSGSRSV1 line 27 and ACCTTYPE
',
     'file' => '',
     'line' => 57,
  )),
  4 => 
  LibXMLError::__set_state(array(
     'level' => 3,
     'code' => 76,
     'column' => 16,
     'message' => 'Opening and ending tag mismatch: OFX line 1 and BANKACCTFROM
',
     'file' => '',
     'line' => 59,
  )),
  5 => 
  LibXMLError::__set_state(array(
     'level' => 3,
     'code' => 5,
     'column' => 1,
     'message' => 'Extra content at the end of the document
',
     'file' => '',
     'line' => 61,
  )),
)

/home/james/workspace/ofxparser/lib/OfxParser/Parser.php:70
/home/james/workspace/ofxparser/lib/OfxParser/Parser.php:51
/home/james/workspace/ofxparser/tests/OfxParser/ParserTest.php:169

ERRORS!
Tests: 28, Assertions: 69, Errors: 1.
```

**Comment by asgrim** [comment]
Also, thanks so much for the updates here, you've inspired me to do a long-overdue update on this library :)

**Comment by ambroisemaupate** [comment]
The error must come from `bpbfc` OFX sample which is still faulty, as I mentionned in my description. This is why I rewrote the tests, to be able to address this issue. *Banque Populaire* OFX file mixes closed tags and unclosed tags so your `Parser` can’t clean the two and it doubles some existing ending tags. See #20 issue from more details.

The unit tests won’t be 100% OK until we find a solution. 😔

**Comment by asgrim** [comment]
Ah - could you remove that fixture for now please, and create a gist or something in the issue #20 please?

**Comment by ambroisemaupate** [comment]
Is it OK if I just comment the line in the data provider?

**Comment by asgrim** [comment]
Okay for now yes, I'll tidy up post-merge. Thanks!

**Comment by ambroisemaupate** [comment]
That’s OK for me now.

```
➜  ofxparser git:(master) ✗ vendor/bin/phpunit tests/
PHPUnit 5.5.2 by Sebastian Bergmann and contributors.

..........................                                        26 / 26 (100%)

Time: 244 ms, Memory: 4.00MB

OK (26 tests, 68 assertions)
```

**Comment by asgrim** [comment]
Great, thank you :)

**Pull request merged**